### PR TITLE
Some SOS fixes/cleanup (bpmd, setclrpath, setsostid).

### PR DESCRIPTION
**Pull request by mikem8361**
Enable the module load notification when no module or methods are found.

setclrpath/setsostid print the current path/tid if no arguments. "setsostid -clear"
now clears the tid/os id mapping.

Added better exception hook to sos plugin to catch special CLRN exception. Before
multiple lldb "Exception breakpoints" where set each time bpmd added a new bp. Added
a "DoNotDeleteOrDisable" message to the exception breakpoint created.

Fix bpmd on release builds. RtlpRaiseException was being inlined and the ExceptionRecord
parameter was being optimized away causing the GetLastEventInformation api to failed
and the special CLRN exception to be ignored.

Add sos help support. "soshelp" displays help about a command.

### CODE REVIEW

**Comment by mikem8361** [comment]
@janvorli could you review? I will have the CentOS build problem fixed soon, but it would be great if you could start.

File: src/ToolBox/SOS/Strike/exts.h [hunk 1]
```diff
@@ -191,13 +191,19 @@ class __ExtensionCleanUp
 
 inline void EENotLoadedMessage(HRESULT Status)
 {
-    ExtOut("Failed to find runtime DLL (clr.dll), 0x%08x\n",Status);
+#ifndef FEATURE_PAL
```
Comment (janvorli) [inline src/ToolBox/SOS/Strike/exts.h]: The dll is called coreclr.dll on Windows too

**Comment by janvorli** [comment]
Other than my comment, it looks good to me.

**Comment by mikem8361** [comment]
Thanks.

**Pull request merged**